### PR TITLE
Added Mecanum Wheel Code

### PR DESCRIPTION
**Pull request by AG-Coding**
This Pull Request closes issue #

I removed a bunch of useless comments and removed a bunch of stuff that we won't use. I also added two subsystems for drive using pid which will create more preciseness
 - 

 - [x] I have labelled this issue with the priority and correct subsystem.
 - [x] I set the person that I worked on this with as a reviewer (who will need to approve the changes)

### CODE REVIEW

**Review by Juliuscease123221 — APPROVED** [review]

**Review by SBsteveyHops — APPROVED** [review]
Looks good, and I do not see any problems.

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -15,11 +17,15 @@ public final class Constants {
...
 
   // Drivetrain
-  public static final double DRIVE_SPEED_MULT = 1.0; // should be between 0 and 1
+  public static final double DRIVE_SPEED_MULT = 1.0;
   public static final double DRIVE_SPEED_AUTO = 0.1 / DRIVE_SPEED_MULT;
   public static final int DRIVE_CURRENT_LIMIT = 40;
   public static final double MOVEMENT_SPEED = 1; // meters per second
   public static final double TURN_SPEED = 1; // radians per second
+  public static final double DIAMETER = 7.9; // Inches
+  public static final double wheel_circumference = Units.inchesToMeters(DIAMETER * Math.PI);
```
Comment (LinusSkucas) [inline src/main/java/frc/robot/Constants.java:26]: ANISH WHAT IS THIS NAMING CONVENSION??????/?????//?>?>?>???

File: src/main/java/frc/robot/Constants.java [hunk 2]
```diff
@@ -15,11 +17,15 @@ public final class Constants {
...
+  public static final double AUTO_WHEELPID_KP = 0.2;
+  public static final double Gearing = 4;
```
Comment (LinusSkucas) [inline src/main/java/frc/robot/Constants.java:28]: it a variable, not no object

**Review by SBsteveyHops — DISMISSED** [review]

File: src/main/java/frc/robot/subsystems/DriveSubsytem.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+  public Pose2d getPose2d() {
+    return poseEstimator.getEstimatedPosition();
+  }
+
+  public MecanumDriveKinematics getKinematics() {
+    return kinematics;
+  }
+
+  public void SetWheelSpeeds(MecanumDriveWheelSpeeds speed, boolean needPID) {
```
Comment (LinusSkucas) [inline src/main/java/frc/robot/subsystems/DriveSubsytem.java:83]:
```suggestion
  public void setWheelSpeeds(MecanumDriveWheelSpeeds speed, boolean needPID) {
```

File: src/main/java/frc/robot/RobotContainer.java [hunk 4]
```diff
@@ -1,53 +1,39 @@
...
+
     drive.setDefaultCommand(
         new RunCommand(
             () ->
-                drive.updateSpeed(
-                    joystickResponse(controller.getRawAxis(0)),
-                    joystickResponse(controller.getRawAxis(1)),
-                    joystickResponse(controller.getRawAxis(3)),
-                    true),
+                drive.setChassisSpeeds(
```
Comment (LinusSkucas) [inline src/main/java/frc/robot/RobotContainer.java:27]: Don't change the joystick input yet

File: src/main/java/frc/robot/subsystems/DriveSubsytem.java [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+import edu.wpi.first.wpilibj.smartdashboard.Field2d;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+
+public class DriveSubsytem extends SubsystemBase {
+
+  private boolean inverted;
+  private ChassisSpeeds ActualChassisSpeeds;
+  private ChassisSpeeds targetChassisSpeeds;
+  private MecanumWheel[] motors = new MecanumWheel[4];
```
Comment (LinusSkucas) [inline src/main/java/frc/robot/subsystems/DriveSubsytem.java:22]: Do not use arrays.

File: src/main/java/frc/robot/subsystems/DriveSubsytem.java [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+  private final MecanumDriveKinematics kinematics;
+  public final WPI_Pigeon2 pigeon;
+  public final Field2d field;
+  private final MecanumDrivePoseEstimator poseEstimator;
+  private final MecanumDriveWheelPositions wheelPositions;
+
+  public DriveSubsytem() {
+    motors[0] = new MecanumWheel(Constants.WHEEL_PORT_FRONT_LEFT, true);
```
Comment (LinusSkucas) [inline src/main/java/frc/robot/subsystems/DriveSubsytem.java:30]: Using arrays in confusing

**Review by LinusSkucas — CHANGES_REQUESTED** [review]

**Review by SBsteveyHops — APPROVED** [review]

**Review by LinusSkucas — APPROVED** [review]

**Review by SBsteveyHops — APPROVED** [review]

**Pull request merged**